homekit/StateManager.js

```diff
@@ -1,623 +1,1039 @@
-	smartModeState.lowTemperatureWebhook = null
-
-	if (device.state.mode === 'COOL') {
-		smartModeState.highTemperatureThreshold = device.state.targetTemperature + (device.usesFahrenheit ? 1.8 : 1)
-		smartModeState.highTemperatureState = {
-			on: true,
-			targetTemperature: device.state.targetTemperature,
-			temperatureUnit: device.temperatureUnit,
-			mode: device.state.mode,
-			fanSpeed: device.state.fanSpeed,
-			swing: device.state.verticalSwing,
-			horizontalSwing: device.state.horizontalSwing,
-			light: device.state.light
+	/**
+	 *
+	 * @param {AirConditioner} device
+	 * @param {boolean} enableClimateReactAutoSetup
+	 */
+	this.updateClimateReact = function(device, enableClimateReactAutoSetup) {
+		if (!enableClimateReactAutoSetup) {
+			return
 		}
 
-		smartModeState.lowTemperatureThreshold = device.state.targetTemperature - (device.usesFahrenheit ? 1.8 : 1)
-		smartModeState.lowTemperatureState = {
-			on: false,
-			targetTemperature: device.state.targetTemperature,
-			temperatureUnit: device.temperatureUnit,
-			mode: device.state.mode,
-			fanSpeed: device.state.fanSpeed,
-			swing: device.state.verticalSwing,
-			horizontalSwing: device.state.horizontalSwing,
-			light: device.state.light
-		}
-	} else if (device.state.mode === 'HEAT') {
-		smartModeState.highTemperatureThreshold = device.state.targetTemperature + (device.usesFahrenheit ? 1.8 : 1)
-		smartModeState.highTemperatureState = {
-			on: false,
-			targetTemperature: device.state.targetTemperature,
-			temperatureUnit: device.temperatureUnit,
-			mode: device.state.mode,
-			fanSpeed: device.state.fanSpeed,
-			swing: device.state.verticalSwing,
-			horizontalSwing: device.state.horizontalSwing,
-			light: device.state.light
-		}
+		// If nothing has changed should we skip...? Like we do in StateHandler for SET?
 
-		smartModeState.lowTemperatureThreshold = device.state.targetTemperature - (device.usesFahrenheit ? 1.8 : 1)
-		smartModeState.lowTemperatureState = {
-			on: true,
-			targetTemperature: device.state.targetTemperature,
-			temperatureUnit: device.temperatureUnit,
-			mode: device.state.mode,
-			fanSpeed: device.state.fanSpeed,
-			swing: device.state.verticalSwing,
-			horizontalSwing: device.state.horizontalSwing,
-			light: device.state.light
+		if (!(device.state instanceof Classes.InternalAcState)) {
+			// TODO: log warning
+			return
 		}
-	}
 
-	// StateHandler is invoked as a Proxy, and therefore overwrites/intercepts the default get()/set() commands [traps]
-	// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy
+		const smartModeState = device.state.smartMode
+
+		smartModeState.type = 'temperature'
+		smartModeState.highTemperatureWebhook = null
+		smartModeState.lowTemperatureWebhook = null
+
+		if (device.state.mode === 'COOL') {
+			smartModeState.highTemperatureThreshold = device.state.targetTemperature + (device.usesFahrenheit ? 1.8 : 1)*climateReactAutoSetupMultiplier + climateReactAutoSetupOffset
```

Comment:
I'd like to understand more about the intention of `climateReactAutoSetupMultiplier`

Reply:
What's the point of `climateReactAutoSetupOffset` if its default is 0 (zero)? How do you expect people to use it and `climateReactAutoSetupMultiplier` together?

To simplify it I'm wondering if the following is better:
```suggestion
			smartModeState.highTemperatureThreshold = device.state.targetTemperature + climateReactAutoSetupOffset
```

Give user complete control over how many degrees above (or below), and have a default of 1 on `climateReactAutoSetupOffset`. Note: the descriptions in README and config schema are (currently) written such that Offset is the main one, Multiplier is placeholder.

We may also need to round the resulting Fahrenheit value as I'm not confident that Sensibo supports decimals on the temp thresholds when in F (like it does in C).